libirods_rule_engine_plugin-indexing.cpp

```diff
@@ -288,23 +301,54 @@ namespace {
                             if(collection_metadata_is_new) {
                                 idx.schedule_collection_operation(
                                     irods::indexing::operation_type::index,
-                                    collection_name,
+                                    logical_path,
                                     _rei->rsComm->clientUser.userName,
                                     value,
                                     units);
                             }
                         }
                     }
-                    if(type == data_object) {
+                    if(type == data_object ||
+                       (type == collection && config->index != attribute)) {
                         idx.schedule_metadata_indexing_event(
-                                collection_name,
+                                logical_path,
                                 _rei->rsComm->clientUser.userName,
                                 attribute,
                                 value,
                                 units);
                     }
                 }
             }
+            else if("pep_api_data_obj_unlink_pre" == _rn) {
+                auto it = _args.begin();
+                std::advance(it, 2);
+                if(_args.end() == it) {
+                    THROW(
+                        SYS_INVALID_INPUT_PARAM,
+                        "invalid number of arguments");
+                }
+                const auto obj_inp = boost::any_cast<dataObjInp_t*>(*it);
+                namespace fs   = irods::experimental::filesystem;
+                namespace fsvr = irods::experimental::filesystem::server;
+                fs::path data_logical_path {obj_inp->objPath};
+                std::string query_str {
+                    boost::str(boost::format("select DATA_ID,META_DATA_ATTR_NAME,META_DATA_ATTR_VALUE,META_DATA_ATTR_UNITS"
+                                             " where DATA_NAME = '%s' and COLL_NAME = '%s'"
+                                             ) % data_logical_path.object_name().string() 
+                                               % data_logical_path.parent_path().string() )
+                };
+                irods::query<rsComm_t> qobj{_rei->rsComm, query_str};
+                for (auto row : qobj) {
```

Comment:
is this supposed to be a copy?

Reply:
It doesn't have to be a copy. Will try as const reference